service/src/test/java/tds/assessment/web/endpoints/ItemControllerIntegrationTests.java

```diff
@@ -0,0 +1,99 @@
+package tds.assessment.web.endpoints;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.util.Optional;
+
+import tds.assessment.ItemFileMetadata;
+import tds.assessment.ItemFileType;
+import tds.assessment.services.ItemService;
+import tds.common.configuration.JacksonObjectMapperConfiguration;
+import tds.common.configuration.SecurityConfiguration;
+import tds.common.web.advice.ExceptionAdvice;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ItemController.class)
+@Import({ExceptionAdvice.class, SecurityConfiguration.class, JacksonObjectMapperConfiguration.class})
+public class ItemControllerIntegrationTests {
+    @MockBean
+    private ItemService mockItemService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private MockMvc http;
+
+    @Test
+    public void shouldReturnStimulusItemMetadata() throws Exception {
+        String url = "/assessments/item/metadata?clientName=SBAC_PT&bankKey=1&stimulusKey=3";
+
+        ItemFileMetadata itemFileMetadata = ItemFileMetadata.create(ItemFileType.STIMULUS, "1-3", "stimulusFile", "stimulusFile/");
+        when(mockItemService.findItemFileMetadataByStimulusKey("SBAC_PT", 1, 3)).thenReturn(Optional.of(itemFileMetadata));
+
+        MvcResult result = http.perform(get(url)
```

Comment:
I like it.  But to test that our API returns as expected, I still think testing using jsonPath is useful.  (e.g. we could configure our ObjectMapper to output any property as any name and round-tripping would work, but the client would break.)

Not requesting a change, but I wanted to throw out a sample controller test from my last project and see what you guys thought:
```
@Test
    public void itShouldCreateABatchComment() throws Exception {
        final Batch batch = createBatch(mock(Vendor.class), UUID.randomUUID().toString());
        when(this.batchService.findOne(eq(batch.getId()))).thenReturn(batch);

        final Comment postedComment = new CommentBuilder()
                .withMessage("This is a comment")
                .build();

        final User authUser = createUser("userId", "A User");
        final Comment persistedComment = createComment(batch, authUser, UUID.randomUUID().toString());
        when(this.commentService.post(any(Comment.class), eq(batch), any(User.class))).thenReturn(persistedComment);
        asUser(authUser);
        try {
            getHttp().perform(post(fromRoot("/batches/" + batch.getId() + "/comments"))
                    .content(getObjectMapper().writeValueAsString(postedComment))
                    .contentType(APPLICATION_JSON)
                    .accept(APPLICATION_JSON))

                    .andExpect(status().is(CREATED.value()))
                    .andExpect(jsonPath("$.id", equalTo(persistedComment.getId())))
                    .andExpect(jsonPath("$.createdDate", instantAs(persistedComment.getCreatedDate())))
                    .andExpect(jsonPath("$.message", equalTo(persistedComment.getMessage())))
                    .andExpect(jsonPath("$.user.name", equalTo(persistedComment.getUser().getName())))
                    .andExpect(jsonPath("$.user.iconUrl", equalTo(persistedComment.getUser().getIconUrl())))
                    .andExpect(jsonPath("$.user.email", equalTo(persistedComment.getUser().getEmail())));

            final ArgumentCaptor<Comment> commentCaptor = ArgumentCaptor.forClass(Comment.class);
            final ArgumentCaptor<User> userCaptor = ArgumentCaptor.forClass(User.class);
            verify(this.commentService).post(commentCaptor.capture(), eq(batch), userCaptor.capture());
            assertThat(commentCaptor.getValue().getMessage()).isEqualTo(postedComment.getMessage());
            assertThat(userCaptor.getValue().getId()).isEqualTo(authUser.getId());
        } finally {
            clearUser();
        }
    }
```